src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -92,7 +92,7 @@ where
                                 "the `--vers` provided, `{}`, is \
                                        not a valid semver version requirement\n\n
                                        Please have a look at \
-                                       http://doc.crates.io/specifying-dependencies.html \
+                                       https://doc.crates.io/specifying-dependencies.html \
```

Comment:
```suggestion
                                       https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html \
```

While you're making the changes can you also point this to the canonical URL?